python-avd/pyavd/_eos_designs/shared_utils/node_type.py

```diff
@@ -23,15 +22,12 @@ class NodeTypeMixin(Protocol):
     """
 
     @cached_property
-    def type(self: SharedUtilsProtocol) -> str:
-        """Type fact set based on type variable."""
+    def type(self: SharedUtilsProtocol) -> str | None:
```

Comment:
this one is potentially a big one.
Also if I have a device_config with a default_node_type matched I am not sure if all the code I have read until now will work :)